clang/lib/Driver/ToolChains/MSVC.cpp

```diff
@@ -131,10 +131,11 @@ void visualstudio::Linker::ConstructJob(Compilation &C, const JobAction &JA,
     CmdArgs.push_back("-defaultlib:oldnames");
   }
 
-  if ((!C.getDriver().IsCLMode() && !Args.hasArg(options::OPT_nostdlib) &&
-       Args.hasArg(options::OPT_fsycl) &&
+  if ((!C.getDriver().IsCLMode() && Args.hasArg(options::OPT_fsycl) &&
        !Args.hasArg(options::OPT_nolibsycl)) ||
       Args.hasArg(options::OPT_fsycl_host_compiler_EQ)) {
+    CmdArgs.push_back(Args.MakeArgString(std::string("-libpath:") +
+                                         TC.getDriver().Dir + "/../lib"));
```

Comment:
Could you add a test that verifies that {{-nostdlib -fsycl}} does not restrict the library?

Reply:
Good point, added tests for `-nostdlib` with `-fsycl` (and also tweaked a `-nolibsycl` test that wasn't properly differentiating between `clang-cl` and `clang++` in the `CHECK:`).